Infrastructure feeds for Java, .NET and Python

bastianeicher opened the pull request:
This add a few "infrastructure" feeds for Java, .NET and Python. Some of these feeds are currently hosted at http://0install.de/feeds/ and some of them are new.

A number of these feeds point to archive locations hosted beneath `http://repo.roscidus.com/`. The following table lists which files would need to be downloaded and reuploaded to the `repo.roscidus.com` hosting in order for the feeds to work.

| Source URI | Destination path |
| --- | --- |
| http://0install.de/files/hosted/jar-launcher-1.0.jar | `java/jar-launcher-1.0.jar` |
| http://0install.de/files/hosted/monopath-emulator.zip | `cli/monopath-emulator.zip` |
| http://0install.de/files/hosted/mono-2.10.9-win32.tar.lzma | `cli/mono-2.10.9-win32.tar.lzma` |
| http://0install.de/files/hosted/pywin32-220.win-amd64-py2.7.tar.bz2 | `python/pywin32-220.win-amd64-py2.7.tar.bz2` |
| http://0install.de/files/hosted/pywin32-220.win-amd64-py3.4.tar.bz2 | `python/pywin32-220.win-amd64-py3.4.tar.bz2` |
| http://0install.de/files/hosted/pywin32-220.win-amd64-py3.6.tar.bz2 | `python/pywin32-220.win-amd64-py3.6.tar.bz2` |
| http://0install.de/files/hosted/pywin32-220.win32-py2.7.tar.bz2 | `python/pywin32-py2.7.tar.bz2` |
| http://0install.de/files/hosted/pywin32-220.win32-py3.4.tar.bz2 | `python/pywin32-220.win32-py3.4.tar.bz2` |
| http://0install.de/files/hosted/pywin32-220.win32-py3.6.tar.bz2 | `python/pywin32-220.win32-py3.6.tar.bz2` |

talex5 commented:
`0repo` says:

> Feed 'feeds/python/python-cario.xml' with uri='http://repo.roscidus.com/python/python-cairo' should be located at 'feeds/python/python-cairo.xml'

(`cario` vs `cairo`)

bastianeicher commented:
Whoops, sorry! Fixed the typo now.

bastianeicher commented:
bump

talex5 commented:
Just to check - are all of these free software according to the FSF definition (https://www.gnu.org/licenses/license-list.html)? The FSF provides the hosting, and this is a condition of that. Not all of them seem to have licenses specified in the feeds.

talex5 commented:
I put the files in `0repo`'s `incoming` directory, changed the archive paths to be relative so that 0repo would handle the uploading, and ran `0repo`. It accepted `gtk.xml`, `python-gtk.xml`, `python-win32.xml`, `python-cairo.xml` and `python-gobject`, which are now published. For `gtk.xml` it complained that there were too many redirections fetching the URL, so I edited it to use the newer `https` versions.

It rejected these for not having a license: `cli-monopath.xml  cli.xml  jar-launcher.xml  mono.xml  netfx.xml`

bastianeicher commented:
All these projects are indeed free software, with the exception of the .NET Framework. However, the feed only lists `<package-implementation>`s for this. There are no archives with binaries involved.

I have rebased my pull request upon the new master and added the missing license attributes to `jar-launcher.xml` and `cli/*.xml`.

talex5 commented on cli/cli-monopath.xml:
```diff
@@ -0,0 +1,153 @@
+        </runner>
+      </command>
+      <manifest-digest sha1new="deb184a5a51f3a7ce3fe633ba188231c29c6aa0d" sha256="3b9225d3133751687c67313bf38208ef63c79f42db2da06661a3ef0dfe177d99" sha256new="HOJCLUYTG5IWQ7DHGE57HAQI55R4PH2C3MW2AZTBUPXQ37QXPWMQ" />
+      <archive href="http://repo.roscidus.com/cli/monopath-emulator.zip" size="11586" type="application/zip" />
```
This archive has the same URL as the others, but a different size.

talex5 commented on java/jar-launcher.xml:
```diff
@@ -0,0 +1,30 @@
+    </command>
+    <implementation version="1.0" stability="stable" id="1.0">
+      <manifest-digest sha1new="3706305d50a745c5a362b02aaf417d2d0765c9c1" sha256="60d4d71585f66a36ac1e3116f234ff93d0319b6d0dccb17e027e162cc9da0481" sha256new="MDKNOFMF6ZVDNLA6GELPENH7SPIDDG3NBXGLC7QCPYLCZSO2ASAQ" />
+      <file href="http://repo.roscidus.com/java/jar-launcher-1.0.jar" size="3201" dest="jar-launcher-1.0.jar" />
```
```
$ 0repo
Processing jar-launcher.xml
Incorrect manifest -- archive is corrupted.
Required digest: sha256new_MDKNOFMF6ZVDNLA6GELPENH7SPIDDG3NBXGLC7QCPYLCZSO2ASAQ
Actual digest: sha256new_2Z754UFCRNQXRRQFVK3K5K7V63ELMWRTWLV7YKL4R6WXT3PWA7WA
```

Manifest is:
```
F 95670c9891a6c9169d61547d795102b9f89c840dff9d13265a6ba2629d0e7b8f 0 3201 jar-launcher-1.0.jar
sha256new_2Z754UFCRNQXRRQFVK3K5K7V63ELMWRTWLV7YKL4R6WXT3PWA7WA
```

bastianeicher commented:
Thanks for catching that! This was a copy-paste blunder. I've now fixed it.

I also made the archive paths relative like you did with the other feeds to allow 0repo handling.

I also fixed the `<file dest=... />` value in `jar-launcher.xml`.

talex5 commented:
OK, I think that's everything now! Thanks :-)
